kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java

```diff
@@ -277,4 +250,32 @@ public void close() throws Exception {
             shutdown();
         }
     }
+
+    @Override
+    public VirtualCluster resolve(String sniHostname, int targetPort) {
```

Comment:
Is `endpointProviders` necessary? We could do the same work with the values of the existing `virtualClusters` map. Could this be a good responsibility for the VirtualCluster itself, if you could ask `boolean matches = virtualCluster.matches(host, port)` and hide this provider work a little?

Reply:
This represents an interim state. In the work I'm planning for #271, an `EndpointRegistry` will assume the responsibility for resolving (port, sniHostname) to a virtual cluster and nodeId.  `ClusterEndpointConfigProvider` will lose its resolution responsibilities.    

I'd like to leave the responsibilities as they are for this PR.